src/schema.ts

```diff
@@ -0,0 +1,385 @@
+	/**
+	 * Which notes represent which channels.
+	 * These cannot be changed via events and thus stay
+	 * the same throughout the course of an entire [[Performance]].
+	 * They can however, be changed while editing a [[Program]].
+	 */
+	notes: { [VC in VibraphoneChannel]: Note };
+}
+
+/**
+ * Any event pertaining to the vibraphone.
+ */
+export type VibraphoneEvent =
+	| VibraphoneVibratoEnabledEvent
+	| VibraphoneVibratoSpeedEvent;
+/** An event corresponding to a change in the [[VibraphoneState.vibratoEnabled]] property. */
+export interface VibraphoneVibratoEnabledEvent {
+	kind: "vibraphone_vibrato_enabled";
+	/** Whether or not the vibrato should be enabled. */
+	vibratoEnabled: boolean;
+}
+/** An event corresponding to a change in the [[VibraphoneState.vibratoSpeed]] property */
+export interface VibraphoneVibratoSpeedEvent {
+	kind: "vibraphone_vibrato_speed";
+	/** The new speed of the vibrato. */
+	vibratoSpeed: number;
+}
+
+/**
+ * Represents a single string on the bass.
+ *
+ * We chose to start at one since musicians count
+ * starting from one and the terminology already
+ * exists in the music realm.
+ */
+export type BassString = 1 | 2 | 3 | 4;
+
+/** Represents the state pertaining to the bass. */
+export interface BassState {
+	/**
+	 * Which frets the capos are on.
+	 * `0` or nothing means no capo is applied to that string.
+	 */
+	capos: { [S in BassString]?: number };
+	/**
+	 * Which notes the strings are tuned to.
+	 * Nothing means the standard bass tuning:
+	 * ```typescript
+	 * {
+	 *   4: "E1",
+	 *   3: "A1",
+	 *   2: "D2",
+	 *   1: "G2",
+	 * }
+	 * ```
+	 */
+	tuning: { [S in BassString]?: Note };
+}
+
+/** Any event pertaining to the bass. */
+export type BassEvent = BassCapoEvent;
+/** An event corresponding to a capo being applied, moved, or removed. */
+export interface BassCapoEvent {
+	kind: "bass_capo";
+	/** Which string is affected. */
+	capoString: BassString;
+	/**
+	 * Which fret is the capo to be applied to.
+	 * `0` or nothing means no capo is applied.
+	 */
+	fret: number;
+}
+
+/** Represents the state pertaining to the hihat machine. */
+export interface HihatMachineState {
+	/** The meaning of this property has yet to be determined. */
+	setting: string;
+}
+
+/** Any event pertaining to the hihat machine. */
+export type HihatMachineEvent = HihatMachineSettingEvent;
+/** An event corresponding to a change in the hihat machine's setting. */
+export interface HihatMachineSettingEvent {
+	kind: "hihatmachine_setting";
+	/**
+	 * The new hihat machine state.
+	 * This will probably change whenever the [[HihatMachineState.setting]]
+	 * is assigned a meaning.
+	 */
+	state: HihatMachineState;
+}
+
+/** Represents the state pertaining to the hihat, not the hihat machine. */
+export interface HihatState {
+	/** Whether or not the hihat is closed. */
+	closed: boolean;
+}
+
+/** Any event pertaining to the hihat, not the hihat machine. */
+export type HihatEvent = HihatClosedEvent;
```

Comment:
May make more sense if changed to `HihatStateEvent` as the open/close state of the hihat is called `HihatState` and not `HihatClosed`

Reply:
With the naming norm we try to tell a bit more about what that event changes. It's always called `...State`, so this might be a very generic name.